File: server/src/main/java/com/heungkuk/academy/domain/survey/entity/Survey.java
package com.heungkuk.academy.domain.survey.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import com.heungkuk.academy.domain.survey.dto.request.SurveyRequest;
import com.heungkuk.academy.global.entity.BaseTimeEntity;
import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Getter
@Entity
@Table(name = "survey")
@Builder
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@AllArgsConstructor(access = AccessLevel.PRIVATE)
public class Survey extends BaseTimeEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "token_id", nullable = false)
    private SurveyToken surveyToken;

    // ── 기본 정보 ──────────────────────────────────────────────
    @Column(nullable = false, length = 50)
    private String location;

    @Column(name = "location_etc", length = 200)
    private String locationEtc;

    @Column(nullable = false, length = 100)
    private String industry;

    @Column(name = "industry_etc", length = 200)
    private String industryEtc;

    @Column(nullable = false, length = 100)
    private String purpose;

    @Column(name = "purpose_etc", length = 200)
    private String purposeEtc;

    @Column(name = "visit_route", nullable = false, length = 50)
    private String visitRoute;

    @Column(name = "visit_route_etc", length = 200)
    private String visitRouteEtc;

    // ── 만족도 (1=매우만족 ~ 5=매우불만족) ──────────────────────
    @Column(name = "staff_service", nullable = false)
    private Integer staffService;

    @Column(name = "staff_service_comment", columnDefinition = "TEXT")
    private String staffServiceComment;

    @Column(nullable = false)
    private Integer cleanliness;

    @Column(name = "cleanliness_comment", columnDefinition = "TEXT")
    private String cleanlinessComment;

    @Column(nullable = false)
    private Integer facilities;

    @Column(name = "facilities_comment", columnDefinition = "TEXT")
    private String facilitiesComment;

    @Column(nullable = false)
    private Integer cafeteria;

    @Column(name = "cafeteria_comment", columnDefinition = "TEXT")
    private String cafeteriaComment;

    @Column(nullable = false)
    private Integer pricing;

    @Column(name = "pricing_comment", columnDefinition = "TEXT")
    private String pricingComment;

    // ── 종합 의견 ──────────────────────────────────────────────
    @Column(nullable = false, length = 100)
    private String revisit;

    @Column(name = "revisit_comment", columnDefinition = "TEXT")
    private String revisitComment;

    @Column(columnDefinition = "TEXT")
    private String comment;

    public static Survey of(SurveyToken surveyToken, SurveyRequest req) {
        return Survey.builder()
                .surveyToken(surveyToken)
                .location(req.getLocation())
                .locationEtc(req.getLocationEtc())
                .industry(req.getIndustry())
                .industryEtc(req.getIndustryEtc())
                .purpose(req.getPurpose())
                .purposeEtc(req.getPurposeEtc())
                .visitRoute(req.getVisitRoute())
                .visitRouteEtc(req.getVisitRouteEtc())
                .staffService(req.getStaffService())
                .staffServiceComment(req.getStaffServiceComment())
                .cleanliness(req.getCleanliness())
                .cleanlinessComment(req.getCleanlinessComment())
                .facilities(req.getFacilities())
                .facilitiesComment(req.getFacilitiesComment())
                .cafeteria(req.getCafeteria())
                .cafeteriaComment(req.getCafeteriaComment())
                .pricing(req.getPricing())
                .pricingComment(req.getPricingComment())
                .revisit(req.getRevisit())
                .revisitComment(req.getRevisitComment())
                .comment(req.getComment())
                .build();
    }
}

File: client/types/survey.ts
export interface SurveyTokenResponse {
  id: number;
  token: string;
  reservationId: string;
}

export interface SurveyResult {
  id: number;
  reservationId: string;
  organization: string | null;
  customer: string | null;
  startDate: string | null;
  endDate: string | null;
  colorCode: string | null;
  createdAt: string;

  location: string;
  locationEtc: string | null;
  industry: string;
  industryEtc: string | null;
  purpose: string;
  purposeEtc: string | null;
  visitRoute: string;
  visitRouteEtc: string | null;

  staffService: number;
  staffServiceComment: string | null;
  cleanliness: number;
  cleanlinessComment: string | null;
  facilities: number;
  facilitiesComment: string | null;
  cafeteria: number;
  cafeteriaComment: string | null;
  pricing: number;
  pricingComment: string | null;

  revisit: string;
  revisitComment: string | null;
  comment: string | null;
}

// 설문 답변 JSON 구조 (11개 문항)
export interface SurveyAnswers {
  // Q1. 회사(단체) 위치
  location: string;
  locationEtc: string;

  // Q2. 업태
  industry: string;
  industryEtc: string;

  // Q3. 연수 목적
  purpose: string;
  purposeEtc: string;

  // Q4. 이용 계기
  visitRoute: string;
  visitRouteEtc: string;

  // Q5. 직원 서비스 만족도 (1=매우만족 ~ 5=매우불만족)
  staffService: number;
  staffServiceComment: string;

  // Q6. 청결 상태 만족도
  cleanliness: number;
  cleanlinessComment: string;

  // Q7. 시설 만족도
  facilities: number;
  facilitiesComment: string;

  // Q8. 식당 음식/서비스 만족도
  cafeteria: number;
  cafeteriaComment: string;

  // Q9. 이용 비용 만족도
  pricing: number;
  pricingComment: string;

  // Q10. 재방문 의향
  revisit: string;
  revisitComment: string;

  // Q11. 자유 의견
  comment: string;
}
